src/steamship/agents/service/agent_service.py

```diff
@@ -338,6 +356,42 @@ def build_default_context(self, context_id: Optional[str] = None, **kwargs) -> A
         context = with_llm(context=context, llm=llm)
         return context
 
+    def _history_file_for_context(self, context_id: Optional[str] = None, **kwargs) -> File:
+        # AgentContexts serve to allow the AgentService to run agents
+        # with appropriate information about the desired tasking.
+        if context_id is None:
+            context_id = uuid.uuid4()
+
+        ctx = AgentContext.get_or_create(self.client, context_keys={"id": f"{context_id}"})
+        return ctx.chat_history.file
+
+    def _streaming_context_id_and_file(
+        self, context_id: Optional[str] = None, **kwargs
+    ) -> Tuple[Optional[str], File]:
+        history_file = self._history_file_for_context(context_id=context_id)
+        ctx_id = context_id
+
+        # if no context ID is provided, try to get a consistent ID from history_file
+        if not ctx_id:
+            ctx_id = _context_key_from_file(key="id", file=history_file)
+
+        # if you can't find a consistent context_id, then something has gone wrong, preventing streaming
+        if not ctx_id:
+            # TODO(dougreid): this points to a slight flaw in the context_keys vs. context_id
+            raise SteamshipError("Error setting up context: no id found for context.")
+
+        return ctx_id, history_file
+
+    @post("async_prompt")
+    def async_prompt(
+        self, prompt: Optional[str] = None, context_id: Optional[str] = None, **kwargs
+    ) -> StreamingResponse:
+        ctx_id, history_file = self._streaming_context_id_and_file(context_id=context_id, **kwargs)
+        task = self.invoke_later(
```

Comment:
I think in a V2 we'll want to follow up and see if we can have the things that are later-invoked auto-block this task from completing (thinking about the comment string about the semantics of the Task below.. we could maybe try to make conformance to that automatic)